src/submit.py

```diff
@@ -43,6 +43,14 @@ async def _check_git_repo(repo_dir: pathlib.Path):
         click.echo(f"Uncommitted changes in {repo_dir}:")
         click.echo(output)
         click.confirm("Are you sure you want to continue?", abort=True)
+        
+    task_info = await HOOKS.getTask()
+    has_internet_access = bool(task_info.permissions)
```

Comment:
I think this would be better cached at pyhooks (or vivaria) rather than in each agent, no?

Reply:
Meanwhile, I assume you mean

`async def write_and_log_instructions(task_info: pyhooks.TaskInfo) -> None:`

I can read that file if you want, but seems like a mistake to me

Reply:
I also don't like that `write_and_log_instructions` writes and logs instructions+permissions rather than just instructions. (noticed because it's unclear how to call the type returned by the function I want to write that reads this file)
This problem could be skipped if we do the caching in pyhooks, wdyt?

Reply:
There is no long-running pyhooks process in which to save a cache, and every interaction with the server carries some risk of timeout or error. The settings are saved in the task environment (see `setup()` in `main.py`). Simply call `settings.get_settings()` to get information about the task and/or agent.